### ISSUE TITLE
Difficulty running SDL examples.

### ISSUE THREAD

**Issue by locallycompact** [HeapsIO/heaps#939]
Hi. I've been trying to run the heaps examples with hashlink on NixOS and I'm running into this issue.

```
[I] lc@aiur ~/S/g/H/h/s/b/blur (master) [1]> haxe blur_hl.hxml
[I] lc@aiur ~/S/g/H/h/s/b/blur (master)> hl blur.hl
src/module.c(352) : FATAL ERROR : Failed to load library sdl.hdll
```

I've had to package this for NixOS, but the libraries should be where they are expected. Indeed it does seem to find fmt.hdll correctly, as I'd had that error whilst I was writing the instructions. My steps are basically the same as the make install:

https://github.com/locallycompact/nixpkgs/commit/ea4cdea11e4fee0145446640321ff304c371325e#diff-ecb72229fe42b1ae59a80e0b8622f9ebe1bdcf3d5ae744edd3b3efac12237d99R22

**Comment by ncannasse** [comment]
You need also SDL2 to be installed on your system

Le dim. 14 mars 2021 à 15:23, locallycompact ***@***.***> a
écrit :

**Comment by locallycompact** [comment]
I do have SDL2 installed and it's included in the dependencies.

**Comment by locallycompact** [comment]
Is there any way I can get a better error message? SDL2 is in my environment, I'm not sure why it wouldn't be able to find it.

**Comment by locallycompact** [comment]
I don't think the presence of SDL2 is the problem since ldd sdl.hdll seems to know about it just fine.

```
libSDL2-2.0.so.0 => /nix/store/k58g6biziz8vpk6zw5n9bym2c6wxjrpd-SDL2-2.0.14/lib/libSDL2-2.0.so.0 (0x00007f6d85c8a000)
```

**Comment by ConstNW** [comment]
looks similar to #467

**Comment by locallycompact** [comment]
I don't think so because I can cause `Failed to load library fmt.hdll` to appear by removing the `cp *.hdll` step and trying to run the result.

**Comment by ncannasse** [comment]
You can look at missing libs with ldd and eventually unbound symbols with nm

Le dim. 14 mars 2021 à 20:13, locallycompact ***@***.***> a
écrit :

**Comment by locallycompact** [comment]
There are no missing libraries and sdl.hdll can find libSDL2.

```
libSDL2-2.0.so.0 => /nix/store/k58g6biziz8vpk6zw5n9bym2c6wxjrpd-SDL2-2.0.14/lib/libSDL2-2.0.so.0 (0x00007f6d85c8a000)
```

**Comment by Ashe** [comment]
I too am now bumping into this issue as a nixos user. Really frustrating as I was so close! I've been enjoying using JS but after having some troubles loading FBX I figured I might try using hashlink instead and im having troubles :(

**Comment by ncannasse** [comment]
Try using "nm" to see if there's some unresolved import.

Le ven. 6 mai 2022 à 22:04, Ashley Smith ***@***.***> a
écrit :

**Comment by Ashe** [comment]
@ncannasse here's the data you requested (including ldd). I ran `nm` and `ldd` on `sdl.hdll`:

https://gist.github.com/Ashe/07bab6b881ffa554813e49d0ff894c72

Hashlink was compiled at version 1.12.

I'm wondering, is it possible this is a wayland issue?

**Comment by locallycompact** [comment]
We now have hashlink working in nixpkgs.
